src/nwb_benchmarks/benchmarks/network_tracking_remote_slicing.py

```diff
@@ -38,16 +39,25 @@ class FsspecNoCacheContinuousSliceBenchmark(BaseNetworkBenchmark):
     object_name = object_name
     slice_range = slice_range
 
-    def setup_cache(self, s3_url: str, object_name: str, slice_range: Tuple[slice]):
-        super().setup_cache(s3_url=self.s3_url, object_name=self.object_name, slice_range=self.slice_range)
-
     def setup(self, s3_url: str, object_name: str, slice_range: Tuple[slice]):
         self.nwbfile, self.io, self.file, self.bytestream = read_hdf5_nwbfile_fsspec_no_cache(s3_url=s3_url)
         self.neurodata_object = get_object_by_name(nwbfile=self.nwbfile, object_name=object_name)
         self.data_to_slice = self.neurodata_object.data
 
-    def operation_to_track_network_activity_of(self, s3_url: str, object_name: str, slice_range: Tuple[slice]):
+    def track_network_statistics_context_proposal(
+        self, s3_url: str, object_name: str, slice_range: Tuple[slice]
+    ) -> Dict[str, Any]:
+        with TrackNetworkStatistics() as network_statistics:
+            self._temp = self.data_to_slice[slice_range]
+        return network_statistics
+
+    def track_network_statistics_manual_proposal(
+        self, s3_url: str, object_name: str, slice_range: Tuple[slice]
+    ) -> Dict[str, Any]:
+        self.start_net_capture()
         self._temp = self.data_to_slice[slice_range]
+        self.stop_netcapture()
+        return self.network_statistics
```

Comment:
Proposal (ii)

Reply:
If the start and stop capture is all we need, then I think it would be cleaner in this proposal to not have these functions on the test class, but instead have them part of a `NetProfiler` class.

```
def track_network_statistics_manual_proposal(
        self, s3_url: str, object_name: str, slice_range: Tuple[slice]
    ) -> Dict[str, Any]:
        nettracker = TrackNetworkStatistics()
        nettracker.start_net_capture()
        self._temp = self.data_to_slice[slice_range]
        nettracker.stop_net_capture()
        return nettracker.network_statistics
```

With this, Proposal i and ii would (I think) be essentially the same. The only difference I think is that the `TrackNetworkStatistics` class in Proposal i needs to have the  additional __enter__ and __exit__ methods.

Reply:
That could work as well; it's whichever method you think is cleaner or easier to implement

The broader question is if you're OK with swapping to only use a single `track_` function that returns the dictionary of information?

Reply:
As long as returning a dict of multiple metrics in one function works with downstream features of ASV, I think that is good solution. The `setup_cache` and multiple tracking function are really more of a work-around , and if I had known that a tracking function can return a dict, I probably would have done it that way, but my understanding of ASV was that each function should return a single metric.

BTW, can we still define the unit for the different measures that are being returned in the dict?

Reply:
> BTW, can we still define the unit for the different measures that are being returned in the dict?

You can still set the attribute (could even set it as a dict of strings for each entry), but there wouldn't be too much outright use for it - I'd think it's clear enough from the entry names what their units ought to be (bytes vs. count)

Reply:
> I'd think it's clear enough from the entry names what their units ought to be

In this case, I would suggest we make sure to just have `in_bytes` `in_seconds` etc. post-fix included in the names to make this explicit.

Reply:
Good point - I've started following that convention myself [in recent data conversions](https://github.com/catalystneuro/visual-coding-to-nwb-v2/blob/main/src/visual_coding_to_nwb_v2/visual_coding_ophys/interfaces/_drifting_grating_stimulus.py#L29) and find it to be very nice